### PR TITLE
Bug 1943614: add enter/exit logging around buildah pull image

### PR DESCRIPTION
**Pull request by gabemontero**
see https://coreos.slack.com/archives/C014MHHKUSF/p1616690484100700

/assign @adambkaplan 
@stevekuznetsov

### CODE REVIEW

File: pkg/build/builder/daemonless.go [hunk 1]
```diff
@@ -193,7 +193,9 @@ func pullDaemonlessImage(sc types.SystemContext, store storage.Store, imageName
 		SystemContext: &systemContext,
 		BlobDirectory: blobCacheDirectory,
 	}
+	log.V(0).Infof("Calling buildah to pull %s", imageName)
```
Comment (adambkaplan) [inline pkg/build/builder/daemonless.go:196]: I don't think we need this - the steps above either fail or succeed quickly.

File: pkg/build/builder/daemonless.go [hunk 2]
```diff
@@ -193,7 +193,9 @@ func pullDaemonlessImage(sc types.SystemContext, store storage.Store, imageName
...
 	_, err = buildah.Pull(context.TODO(), "docker://"+imageName, options)
+	log.V(0).Infof("Returned from calling buildah to pull %s with error received %v", imageName, err != nil)
```
Comment (adambkaplan) [inline pkg/build/builder/daemonless.go:198]: I don't think we need to log the error here - if one is raised, we'll log it elsewhere.

```suggestion
	log.V(0).Infof("Finished pulling image %s", imageName)
```

**Review by adambkaplan — CHANGES_REQUESTED** [review]

**Comment by gabemontero** [comment]
I believe your comments on the new logging for the test platform team have been addressed @adambkaplan ptal

created tracking bz @stevekuznetsov as I'm assuming you all need this in 4.7 for your current triage

/cherrypick release-4.7

**Comment by gabemontero** [comment]
unhealthy cluster with image eco ... connection refused on api calls

/test e2e-aws-image-ecosystem

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Pull request merged**